planner/core/memtable_predicate_extractor.go

```diff
@@ -661,6 +790,235 @@ func (e *ClusterLogTableExtractor) explainInfo(p *PhysicalMemTable) string {
+	// e.g: SELECT * FROM tidb_hot_regions_history WHERE db_name in ('mysql', 'test')
+	DBNames set.StringSet
+	// TableNames represents table names applied to, and we should apply all tables' hot regions if there is no table specified.
+	// e.g: SELECT * FROM tidb_hot_regions_history WHERE table_name in ('tables_priv', 'stats_meta')
+	TableNames set.StringSet
+	// IndexNames represents index names applied to, and we should apply all indexes' hot regions if there is no index specified.
+	// e.g: SELECT * FROM tidb_hot_regions_history WHERE index_name in ('idx_ver', 'tbl')
+	IndexNames set.StringSet
+	// TableID(IndexID) represents all table(index) ids we should filter in TiDB
+	// e.g:
+	// 1. SELECT * FROM tidb_hot_regions_history WHERE table_id=11
+	// 2. SELECT * FROM tidb_hot_regions_history WHERE table_id in (11, 21)
+	TableIDs []uint64
+	IndexIDs []uint64
+}
+
+// Extract implements the MemTablePredicateExtractor Extract interface
+func (e *HotRegionsHistoryTableExtractor) Extract(
+	ctx sessionctx.Context,
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+) []expression.Expression {
+	// Extract the `region_id/store_id/peer_id/type/db_name/table_name/index_name/table_id/index_id` columns
+	remained, regionIDSkipRequest, regionIDs := e.extractCol(schema, names, predicates, "region_id", false)
+	remained, storeIDSkipRequest, storeIDs := e.extractCol(schema, names, remained, "store_id", false)
+	remained, peerIDSkipRequest, peerIDs := e.extractCol(schema, names, remained, "peer_id", false)
+	remained, typeSkipRequest, types := e.extractCol(schema, names, remained, "type", false)
+	remained, dbNameSkipRequest, dbNames := e.extractCol(schema, names, remained, "db_name", true)
+	remained, tableNameSkipRequest, tableNames := e.extractCol(schema, names, remained, "table_name", true)
+	remained, indexNameSkipRequest, indexNames := e.extractCol(schema, names, remained, "index_name", true)
+	remained, tableIDSkipRequest, tableIDs := e.extractCol(schema, names, remained, "table_id", false)
+	remained, indexIDSkipRequest, indexIDs := e.extractCol(schema, names, remained, "index_id", false)
+
+	e.SkipRequest = regionIDSkipRequest || storeIDSkipRequest || peerIDSkipRequest ||
+		typeSkipRequest || dbNameSkipRequest || tableNameSkipRequest ||
+		indexNameSkipRequest || tableIDSkipRequest || indexIDSkipRequest
+	if e.SkipRequest {
+		return nil
+	}
+	e.RegionIDs, e.StoreIDs, e.PeerIDs = e.parseQuantilesUint64(regionIDs), e.parseQuantilesUint64(storeIDs), e.parseQuantilesUint64(peerIDs)
+	e.HotRegionTypes, e.DBNames, e.TableNames, e.IndexNames = types, dbNames, tableNames, indexNames
+	e.TableIDs, e.IndexIDs = e.parseQuantilesUint64(tableIDs), e.parseQuantilesUint64(indexIDs)
+
+	remained, startTime, endTime := e.extractTimeRange(ctx, schema, names, remained, "update_time", time.Local)
+	// The time unit for search hot regions is millisecond.
```

Comment:
extractTimeRange function can solve ms currently, / 1000 before send to PD because PD use sec unit.